PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/HardcodedAPIKeysAndTokensCheck.java

```diff
@@ -0,0 +1,116 @@
+ *     }
+ */
+public class HardcodedAPIKeysAndTokensCheck extends LocalInspectionTool {
+
+    @NotNull
+    @Override
+    public PsiElementVisitor buildVisitor(@NotNull ProblemsHolder holder, boolean isOnTheFly) {
+        return new APIKeysAndTokensVisitor(holder, isOnTheFly);
+    }
+
+    // This class is a visitor that checks for the use of specific methods that authenticate API keys and tokens.
+    public static class APIKeysAndTokensVisitor extends JavaElementVisitor {
+
+        private final ProblemsHolder holder;
+        private final boolean isOnTheFly;
+
+        // Load rule configurations from a JSON file
+        static Map<String, Object> RULE_CONFIGURATIONS = loadRuleConfigurations();
+        static String ANTI_PATTERN_MESSAGE = (String) RULE_CONFIGURATIONS.get("anti_pattern_message");
+        static List<String> CLIENTS_TO_CHECK = (List<String>) RULE_CONFIGURATIONS.get("clients_to_check");
+
+        // This constructor is used to create a visitor for the inspection
+        public APIKeysAndTokensVisitor(ProblemsHolder holder, boolean isOnTheFly) {
+            this.holder = holder;
+            this.isOnTheFly = isOnTheFly;
+        }
+
+        @Override
+        public void visitElement(@NotNull PsiElement element) {
+            super.visitElement(element);
+
+            // Check if the element is a new expression -- i.e., a constructor call
+            if (element instanceof PsiNewExpression) {
+
+                // Cast the element to a new expression
+                PsiNewExpression newExpression = (PsiNewExpression) element;
+
+                // Get the class reference name from the new expression
+                String classReference = newExpression.getClassReference().getReferenceName();
+
+                // Check if the class reference is not null, the qualifier name starts with "com.azure" and
+                // the class reference is in the list of clients to check
+                if (newExpression.getClassReference() != null
+                        && newExpression.getClassReference().getQualifiedName().startsWith("com.azure")
+                        && CLIENTS_TO_CHECK.contains(classReference)) {
+                    this.holder.registerProblem(newExpression, ANTI_PATTERN_MESSAGE);
+                }
+            }
+        }
+
+
+        // Helper method to load rule configurations from a JSON file
+        private static Map<String, Object> loadRuleConfigurations() {
+
+            // Define constants for string literals
+            final String RULE_KEY = "HardcodedAPIKeysAndTokensCheck";
+            final String ANTI_PATTERN_MESSAGE_KEY = "anti_pattern_message";
+            final String CLIENTS_TO_CHECK_KEY = "clients_to_check";
+            final String RULE_CONFIGURATION = "META-INF/ruleConfigs.json";
+
+            // Load the rule configurations from the JSON file
+            try {
```

Comment:
This section can be updated to ensure input stream is properly closed as well.

```java
try (InputStream inputStream = HardcodedAPIKeysAndTokensCheck.class.getClassLoader().getResourceAsStream(RULE_CONFIGURATION)) {
    if (inputStream == null) {
        throw new FileNotFoundException("Configuration file not found");
    }
    // Read the JSON object from the input stream
    try (BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(inputStream))) {
        final JSONObject jsonObject = new JSONObject(bufferedReader.lines().collect(Collectors.joining()));
        final JSONObject apiKeysAndTokensObject = jsonObject.getJSONObject(RULE_KEY);
        final String antiPatternMessage = apiKeysAndTokensObject.getString(ANTI_PATTERN_MESSAGE_KEY);
        final List<String> clientsToCheck = apiKeysAndTokensObject.getJSONArray(CLIENTS_TO_CHECK_KEY).toList().stream().map(Object::toString).collect(Collectors.toList());
        return Map.of(ANTI_PATTERN_MESSAGE_KEY, antiPatternMessage, CLIENTS_TO_CHECK_KEY, clientsToCheck);
    }
} catch (Exception e) {
    e.printStackTrace();
    return Collections.emptyMap();
}
```

Reply:
Thanks! I also refactored this section in a separate PR so I will be removing the json loading functionality from this file: https://github.com/Njeriiii/azure-tools-for-java/pull/16/files#diff-e739175547b0efaff446f64bf6c8f34d16e4e14f275e75ebd38e47a677b0e1b9R84